h2/src/main/org/h2/store/fs/FilePathDisk.java

```diff
@@ -43,6 +45,23 @@ public FilePathDisk getPath(String path) {
 
     @Override
     public long size() {
+        if (name.startsWith(CLASSPATH_PREFIX)) {
+            try {
+                String fileName = name.substring(CLASSPATH_PREFIX.length());
+                // Force absolute resolution in Class.getResource
+                if (!fileName.startsWith("/")) {
+                    fileName = "/" + fileName;
+                }
+                URL resource = this.getClass().getResource(fileName);
+                if (resource != null) {
+                    return Files.size(Paths.get(this.getClass().getResource(fileName).toURI()));
```

Comment:
why not just `return Files.size(Paths.get(resource.toURI()));` ?

Reply:
Thanks for the hint, I've just updated the pull request.